aws-api/src/main/java/com/amplifyframework/api/aws/MultiAuthAppSyncGraphQLOperation.java

```diff
@@ -0,0 +1,250 @@
+ */
+
+package com.amplifyframework.api.aws;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.api.ApiException;
+import com.amplifyframework.api.ApiException.ApiAuthException;
+import com.amplifyframework.api.aws.auth.ApiRequestDecoratorFactory;
+import com.amplifyframework.api.aws.auth.RequestDecorator;
+import com.amplifyframework.api.graphql.GraphQLOperation;
+import com.amplifyframework.api.graphql.GraphQLRequest;
+import com.amplifyframework.api.graphql.GraphQLResponse;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.core.model.auth.AuthorizationTypeIterator;
+import com.amplifyframework.logging.Logger;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+/**
+ * An operation to enqueue a GraphQL request to OkHttp client,
+ * with the goal of obtaining a list of responses. For example,
+ * this is used for a LIST query vs. a GET query or most mutations.
+ * @param <R> Casted type of GraphQL result data
+ */
+public final class MultiAuthAppSyncGraphQLOperation<R> extends GraphQLOperation<R> {
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-api");
+    private static final String CONTENT_TYPE = "application/json";
+    private static final String UNAUTHORIZED_EXCEPTION = "UnauthorizedException";
+
+    private final String endpoint;
+    private final OkHttpClient client;
+    private final Consumer<GraphQLResponse<R>> onResponse;
+    private final Consumer<ApiException> onFailure;
+    private final ApiRequestDecoratorFactory apiRequestDecoratorFactory;
+    private final ExecutorService executorService;
+
+    private AuthorizationTypeIterator authTypes;
+    private Call ongoingCall;
+
+    /**
+     * Constructs a new AppSyncGraphQLOperation.
+     * @param builder An instance of the {@link Builder} object.
+     */
+    private MultiAuthAppSyncGraphQLOperation(Builder<R> builder) {
+        super(builder.request, builder.responseFactory);
+        this.apiRequestDecoratorFactory = builder.apiRequestDecoratorFactory;
+        this.endpoint = builder.endpoint;
+        this.client = builder.client;
+        this.onResponse = builder.onResponse;
+        this.onFailure = builder.onFailure;
+        this.executorService = builder.executorService;
+
+        if (!(getRequest() instanceof AppSyncGraphQLRequest)) {
+            onFailure.accept(new ApiException(
+                "Multiauth only supported with AppSyncGraphQLRequest<T>.",
+                AmplifyException.TODO_RECOVERY_SUGGESTION
+            ));
+            return;
+        }
```

Comment:
feels like a weird pattern (triggering error callback in constructor). can we just accept `AppSyncGraphQLRequest` in the builder?

Reply:
Addressed by passing the request into the builder.